biz.aQute.bndlib.tests/src/test/ClazzTest.java

```diff
@@ -382,4 +391,249 @@ public void testMultipleInstructions() throws Exception {
 			assertTrue(clazz.is(QUERY.NAMED, new Instruction("!junit.framework.*"), analyzer));
 		}
 	}
+
+	public static interface Foo<T> {}
+
+	public class GenericMethodParameters {
+		void bindChars(Foo<Character> c) {}
+	}
+
+	public void testGenericMethodParameters() throws Exception {
+		File file = IO.getFile("bin/test/ClazzTest$GenericMethodParameters.class");
+		try (Analyzer analyzer = new Analyzer()) {
+			Clazz clazz = new Clazz(analyzer, file.getPath(), new FileResource(file));
+			clazz.parseClassFileWithCollector(new ClassDataCollector() {
+				@Override
+				public void method(MethodDef methodDef) {
+					if (methodDef != null && "bindChars".equals(methodDef.getName())) {
+						String signature = methodDef.getSignature();
+						if (signature == null) {
```

Comment:
While this is true for this specific case, is it true in general that signature is normalized when there are no generics? Because my experience was that signature was only non-null when generics were part of the signature, otherwise only the Descriptor was non-null.

Reply:
Correct. Generally, unless generics were used, there will be no signature attribute for the element and this signature will be null. So for methods, you fall back to the description. For fields, the description may not always be a valid signature since fields can be of non-reference types.